src/compute.rs

```diff
@@ -169,3 +174,8 @@ pub fn merkle_root(data: &[u8]) -> [u8; 32] {
 
     hasher.finalize().into()
 }
+
+/// logical xor
+pub fn xor(a: bool, b: bool) -> bool {
+    (a && !b) || (b && !a)
+}
```

Comment:
Can't we just use `^` here? https://doc.rust-lang.org/reference/types/boolean.html#logical-xor

Reply:
I mean, can't we replace this function altogether?

Reply:
Yeah, `BitXor` is fine here.  I tried it at one point before I got everything else working, and erroneously thought that it was the problem.  I just tried it again and it's good.

Reply:
Fixed